src/DependencyInjection/Compiler/RegisterHookListenersPass.php

```diff
@@ -56,38 +56,47 @@ private function getHooks(ContainerBuilder $container): array
 
         foreach ($serviceIds as $serviceId => $tags) {
             foreach ($tags as $attributes) {
-                $this->checkRequiredAttributes($serviceId, $attributes);
-
-                $priority = (int) ($attributes['priority'] ?? 0);
-                $hook = $attributes['hook'];
-
-                $hooks[$hook][$priority][] = [$serviceId, $attributes['method']];
+                $this->addHookCallback($hooks, $serviceId, $attributes);
             }
         }
 
         return $hooks;
     }
 
     /**
-     * Checks that required attributes (hook and method) are set.
+     * Adds hook for given service and attributes.
      *
      * @param string $serviceId
      * @param array  $attributes
      *
      * @throws InvalidConfigurationException
      */
-    private function checkRequiredAttributes(string $serviceId, array $attributes): void
+    private function addHookCallback(array &$hooks, string $serviceId, array $attributes): void
     {
         if (!isset($attributes['hook'])) {
             throw new InvalidConfigurationException(
                 sprintf('Missing hook attribute in tagged hook service with service id "%s"', $serviceId)
             );
         }
 
-        if (!isset($attributes['method'])) {
-            throw new InvalidConfigurationException(
-                sprintf('Missing method attribute in tagged hook service with service id "%s"', $serviceId)
-            );
+        $priority = (int) ($attributes['priority'] ?? 0);
+
+        $hooks[$attributes['hook']][$priority][] = [$serviceId, $this->getMethod($attributes)];
+    }
+
+    /**
+     * Gets the method name from config or hook name.
+     *
+     * @param array $attributes
+     *
+     * @return string
+     */
+    private function getMethod(array $attributes): string
+    {
+        if (isset($attributes['method'])) {
+            return (string) $attributes['method'];
```

Comment:
Why did you cast to string here?

Reply:
why? It could be an integer, can it not? Just making sure the return type of the function is met…